core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -391,6 +391,41 @@ private[spark] class AppStatusStore(
     ordered.skip(offset).max(length).asScala.map(_.toApi).toSeq
   }
 
+  // Filters task list based on search parameter
+  def filterTaskList(
```

Comment:
this function doesn't really have anything to do with the store, its simply filtering on the taskData so I think it should just live in StagesResource.scala

Reply:
Done